conans/test/functional/compression/compression_test.py

```diff
@@ -0,0 +1,443 @@
+            for zi in f.infolist():
+                if (zi.external_attr >> 28) == 0xA:
+                    f.extract(zi, path=d)
+                    dst_name = os.path.join(d, zi.filename)
+                    data = load(dst_name)
+                    os.unlink(dst_name)
+                    create_symlink(data, dst_name)
+                else:
+                    f.extract(zi, path=d)
+                    dt = time.mktime(zi.date_time + (0, 0, -1))
+                    os.utime(os.path.join(d, zi.filename), (dt, dt))
+                    if os.name == 'posix':
+                        permissions = zi.external_attr >> 16
+                        os.chmod(os.path.join(d, zi.filename), permissions)
+            f.close()
+
+
+class TarCompressor(Compressor):
+    read_mode = "r"
+    write_mode = "w"
+    ext = ".tar"
+
+    def compress(self, d):
+        with tarfile.open(name='test' + self.ext, mode=self.write_mode, format=tarfile.GNU_FORMAT) as f:
+            inodes = dict()
+            for root, dirnames, filenames in os.walk(d):
+                for filename in filenames:
+                    filename = os.path.join(root, filename)
+                    archname = os.path.relpath(filename, d)
+
+                    ti = tarfile.TarInfo(archname)
+                    if os.path.islink(filename):
+                        ti.type = tarfile.SYMTYPE
+                        ti.linkname = os.readlink(filename)
+                        ti.size = 0
+                        f.addfile(tarinfo=ti)
+                    else:
+                        st = os.stat(filename)
+                        if st.st_ino in inodes:
+                            ti.type = tarfile.LNKTYPE
+                            ti.linkname = inodes[st.st_ino]
+                            ti.size = 0
+                            f.addfile(tarinfo=ti)
+                        else:
+                            inodes[st.st_ino] = archname
+                            ti.size = st.st_size
+                            ti.mode = st.st_mode
+                            ti.mtime = st.st_mtime
+                            with open(filename, 'rb') as file_handler:
+                                f.addfile(tarinfo=ti, fileobj=file_handler)
+                for dirname in dirnames:
+                    filename = os.path.join(root, dirname)
+                    archname = os.path.relpath(filename, d)
+                    ti = tarfile.TarInfo(archname)
+                    if is_link(filename):
+                        ti.type = tarfile.SYMTYPE
+                        ti.linkname = os.readlink(filename)
+                        ti.size = 0
+                        f.addfile(tarinfo=ti)
+            f.close()
+
+    def decompress(self, d):
+        import tarfile
+        with tarfile.open(name='test' + self.ext, mode=self.read_mode) as f:
+            for ti in f.getmembers():
+                if ti.issym():
+                    filename = os.path.join(d, ti.name)
```

Comment:
What happens if we do directly a ``extractall()``?

Reply:
I remember I didn't manage to get it work correctly, but right now I see it's passing for me with `extractall()` (at least on Windows). I'll do additional tests to double-check, if it works correctly, I'll just leave `extractall`.

Reply:
replaced to `extractall` - hope CI passes as well

Reply:
It is also great to have negative tests. If ``extractall`` fails in some scenario, lets test that scenario and check for failure. In this way everything is documented and we will not think again in the future "why is it that this is complicated and we are not using something simpler as ``extractall``?

Reply:
at least for tar I hope it works out of the box gracefully. for zip, it doesn't handle symlinks at all, and everyone advices to handle in code.